go.mod

```diff
@@ -1,59 +1,60 @@
 module github.com/kanopy-platform/gateway-certificate-controller
 
-go 1.20
+go 1.23
 
 require (
 	github.com/cert-manager/cert-manager v1.8.0
-	github.com/go-logr/logr v1.2.3
-	github.com/prometheus/client_golang v1.12.1
-	github.com/sirupsen/logrus v1.8.1
-	github.com/spf13/cobra v1.4.0
+	github.com/go-logr/logr v1.4.1
+	github.com/prometheus/client_golang v1.18.0
+	github.com/sirupsen/logrus v1.9.0
+	github.com/spf13/cobra v1.7.0
 	github.com/spf13/viper v1.10.0
-	github.com/stretchr/testify v1.7.0
-	go.uber.org/zap v1.21.0
-	istio.io/api v0.0.0-20220512212136-561ffec82582
-	istio.io/client-go v1.13.4
-	k8s.io/api v0.24.0
-	k8s.io/apimachinery v0.24.0
-	k8s.io/cli-runtime v0.24.0
-	k8s.io/client-go v0.24.0
-	sigs.k8s.io/controller-runtime v0.12.1
+	github.com/stretchr/testify v1.8.4
+	go.uber.org/zap v1.26.0
+	istio.io/api v1.21.6
+	istio.io/client-go v1.21.6
+	k8s.io/api v0.29.2
+	k8s.io/apimachinery v0.29.2
+	k8s.io/cli-runtime v0.29.2
+	k8s.io/client-go v0.29.2
+	sigs.k8s.io/controller-runtime v0.17.2
```

Comment:
annoyingly controller-runtime debug logs for webhook do not appear anymore due to this change https://github.com/kubernetes-sigs/controller-runtime/pull/2334/files

```
[kanopy-gateway-cert-controller] {"level":"debug","ts":1733927301.1292105,"logger":"controller-runtime.webhook.webhooks","msg":"received request","webhook":"/mutate","UID":"c910ccfb-0da6-4f9a-a2cd-d93aba21bb8c","kind":"networking.istio.io/v1beta1, Kind=Gateway","resource":{"group":"networking.istio.io","version":"v1beta1","resource":"gateways"}}
[kanopy-gateway-cert-controller] {"level":"info","ts":1733927301.1294222,"msg":"mutating gateway httpbin-gateway Tls.CredentialName,  to default-httpbin-gateway-https"}
[kanopy-gateway-cert-controller] {"level":"debug","ts":1733927301.1296973,"logger":"controller-runtime.webhook.webhooks","msg":"wrote response","webhook":"/mutate","code":200,"reason":"","UID":"c910ccfb-0da6-4f9a-a2cd-d93aba21bb8c","allowed":true}
```

The 1st and 2nd line do not appear anymore with --log-level=debug

Reply:
It looks like we need to explicitly set the Logger on the management service now. 

```
		},
		Logger: klog.Log,
```

Reply:
Added but still no debug logs

Reply:
I get debug level info with that enabled.  I don't know why the webhook handler doesn't link up with that. but as far as general debug logs; this change seems necessary.

```[kanopy-gateway-cert-controller] {"level":"debug","ts":"2024-12-12T19:40:36Z","msg":"Debug","controller":"istio-gateway-controller","object":{"name":"httpbin-gateway","namespace":"devops"},"namespace":"devops","name":"httpbin-gateway","reconcileID":"43cf7199-0f8f-4dfe-b14c-013f6c4855b2"}
```

Reply:
You'll see Debug logs if controller-runtime uses .V(1).
However for [this log](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.17.2/pkg/webhook/admission/http.go#L117) they've picked .V(5) so we have to set zapcore.Level(-5) in my latest update.

Now the logs show but the `"level":"Level(-5)"` is ugly
```
[kanopy-gateway-cert-controller] {"level":"Level(-5)","ts":"2024-12-12T22:14:01Z","logger":"admission","msg":"received request","object":{"name":"httpbin-gateway","namespace":"devops"},"namespace":"devops","name":"httpbin-gateway","resource":{"group":"networking.istio.io","version":"v1beta1","resource":"gateways"},"user":"kubernetes-admin","requestID":"5d0dc056-7e75-4689-96b7-fce885beb918"}
[kanopy-gateway-cert-controller] {"level":"info","ts":"2024-12-12T22:14:01Z","logger":"admission","msg":"mutating gateway httpbin-gateway Tls.CredentialName,  to devops-httpbin-gateway-https","object":{"name":"httpbin-gateway","namespace":"devops"},"namespace":"devops","name":"httpbin-gateway","resource":{"group":"networking.istio.io","version":"v1beta1","resource":"gateways"},"user":"kubernetes-admin","requestID":"5d0dc056-7e75-4689-96b7-fce885beb918"}
[kanopy-gateway-cert-controller] {"level":"Level(-5)","ts":"2024-12-12T22:14:01Z","logger":"admission","msg":"wrote response","code":200,"reason":"","message":"","requestID":"5d0dc056-7e75-4689-96b7-fce885beb918","allowed":true}
```

If they change the .V() verbosity to higher, then we'd need to update again.

Reply:
Thats ok; we shouldn't be running with debug by default in prod It is mostly a dev flow or if we need to conditionally turn this on in other environments.